src/Ipac_Table_Writer/internals.cxx

```diff
@@ -92,6 +151,31 @@ void tablator::Ipac_Table_Writer::write_consecutive_records(
 
 /**********************************************************/
 
+void tablator::Ipac_Table_Writer::write_consecutive_records(
+        const Table& table, std::ostream& os,
+        const std::vector<size_t>& included_column_ids, size_t start_row,
+        size_t row_count, const std::vector<Data_Type>& datatypes_for_writing) {
+    size_t num_table_rows = table.num_rows();
+    if (start_row > num_table_rows) {
+        std::string msg = "Invalid start_row ";
+        msg.append(std::to_string(start_row))
+                .append("; only")
+                .append(std::to_string(num_table_rows))
+                .append(" rows in table.");
+        throw std::runtime_error(msg);
+    }
```

Comment:
`start_row >= num_table_rows`? Also, consider not doing anything instead of throwing an exception? Invalid row IDs are skipped when a vector of row indexes is supplied, so behavior when indexes are out of range is somewhat inconsistent at the moment.

Reply:
Ah, okay, I will silently skip both invalid columns and invalid rows.
I have mixed feelings about bounds checks.  Do them up front and exit as soon as possible?  Let the innermost functions handle them so we aren't repeating the checks in every single outward-facing function?  Have all functions handle them for extra security?